src/outputs/alsa.c

```diff
@@ -168,47 +174,46 @@ alsa_session_cleanup(struct alsa_session *as)
 static struct alsa_session *
 alsa_session_make(struct output_device *device, output_status_cb cb)
 {
-  struct output_session *os;
   struct alsa_session *as;
 
-  os = calloc(1, sizeof(struct output_session));
-  if (!os)
-    {
-      DPRINTF(E_LOG, L_LAUDIO, "Out of memory for ALSA session (os)\n");
-      return NULL;
-    }
-
   as = calloc(1, sizeof(struct alsa_session));
   if (!as)
     {
       DPRINTF(E_LOG, L_LAUDIO, "Out of memory for ALSA session (as)\n");
-      free(os);
-      return NULL;
+      goto failure_cleanup;
```

Comment:
Here you can just return? You can also change it to use the CHECK_NULL(L_LAUDIO, as = calloc(...)), it is fine for OOM.

Reply:
OK. This is one of those philosophical things. I saw that too when I wrote it, that I could just return.  I tend to have a style of trying to make cleanup code handle all cases, and to force all errors through the same cleanup code path.  It makes it easier to add more code later without having to change existing code. However, maybe this is more distracting than helpful. I don't have a strong feeling about it. I am content to change it.